types/web3-provider-engine.d.ts

```diff
@@ -13,8 +13,8 @@ declare module 'web3-provider-engine' {
     jsonrpc: Y['jsonrpc'];
     result: V | undefined;
     error?: {
-      message: unknown;
```

Comment:
This fixes the casting that we need to do in `create-network-client`. It also makes it possible to have our `buildFakeProvider` helper in the unit tests return the same type as the provider that NetworkController uses and makes this type compatible with `eth-query`.